types/react-redux/react-redux-tests.tsx

```diff
@@ -93,6 +91,16 @@ function MapDispatch() {
 
     const verifyNull = <TestNull foo='bar' />;
 
+    // We deliberately cast the right-hand side to any because…
+    const mapDispatchToProps2: MapDispatchToProps<DispatchProps, OwnProps> = {} as any;
```

Comment:
I need to finish writing this comment, but truth is, I don't understand why.

If 2 values have the same type annotation but different values on the right-hand side (on none at all if it's a `declare`), TS seems to behave differently. This is essentially the behaviour we're trying to test for here. I don't know why TS does this, nor what it's called. Can anyone enlighten me? Then I will finish writing my comment about why this test is needed in addition to the one above.

``` ts
type MyUnion<T> = (() => T) | T;

const value1: MyUnion<{ foo: number }> = { foo: 1 };
// No type error
value1.foo;

const value2: MyUnion<{ foo: number }> = 1 as any;
// Type error
value2.foo;

declare const value3: MyUnion<{ foo: number }>
// Type error
value3.foo;
```